src/BuiltInTools/HotReloadAgent/HotReloadAgent.cs

```diff
@@ -195,4 +198,40 @@ private void ApplyDeltas(Assembly assembly, IReadOnlyList<UpdateDelta> deltas)
 
         return default;
     }
+
+    /// <summary>
+    /// Applies the content update.
+    /// </summary>
+    public void ApplyStaticAssetUpdate(StaticAssetUpdate update)
+    {
+        _metadataUpdateHandlerInvoker.ContentUpdated(update);
+    }
+
+    /// <summary>
+    /// Clear any hot-reload specific environment variables. This prevents child processes from being
+    /// affected by the current app's hot reload settings. See https://github.com/dotnet/runtime/issues/58000
+    /// </summary>
+    public static void ClearHotReloadEnvironmentVariables(Type startupHookType)
+    {
+        Environment.SetEnvironmentVariable(AgentEnvironmentVariables.DotNetStartupHooks,
+            RemoveCurrentAssembly(startupHookType, Environment.GetEnvironmentVariable(AgentEnvironmentVariables.DotNetStartupHooks)!));
+
+        Environment.SetEnvironmentVariable(AgentEnvironmentVariables.DotNetWatchHotReloadNamedPipeName, "");
+        Environment.SetEnvironmentVariable(AgentEnvironmentVariables.HotReloadDeltaClientLogMessages, "");
+    }
+
+    // internal for testing
+    internal static string RemoveCurrentAssembly(Type startupHookType, string environment)
+    {
+        if (environment is "")
+        {
+            return environment;
+        }
+
+        var assemblyLocation = startupHookType.Assembly.Location;
+        var updatedValues = environment.Split(Path.PathSeparator, StringSplitOptions.RemoveEmptyEntries)
+            .Where(e => !string.Equals(e, assemblyLocation, StringComparison.OrdinalIgnoreCase));
```

Comment:
should this comparison depend on the string comparison rules for the file system?

Reply:
Code just moved. But yeah, it should probably do ordinal on Linux.